checks.d/hdfs_namenode.py

```diff
@@ -0,0 +1,193 @@
+        jmx_address = instance.get('hdfs_namenode_jmx_uri', JMX_URI)
+
+        # Get metrics from JMX
+        self._hdfs_namenode_metrics(jmx_address, JMX_PATH)
+
+    def _hdfs_namenode_metrics(self, jmx_uri, jmx_path):
+        '''
+        Get HDFS data node metrics from JMX
+        '''
+        response = self._rest_request_to_json(jmx_uri,
+            jmx_path,
+            JMX_SERVICE_CHECK,
+            qry='Hadoop:service=NameNode,name=FSNamesystemState')
+
+        beans = response.get('beans', [])
+
+        for bean in beans:
+            bean_name = bean.get('name')
+
+            if bean_name is not None \
+                    and bean_name == HDFS_NAMENODE_BEAN_NAME:
+
+                for metric, (metric_name, metric_type) in HDFS_METRICS.iteritems():
+                    metric_value = bean.get(metric)
+
+                    if metric_value is not None:
+                        self._set_metric(metric_name, metric_type, metric_value)
+
+    def _set_metric(self, metric_name, metric_type, value, tags=None, device_name=None):
+        '''
+        Set a metric
+        '''
+        if metric_type == GAUGE:
+            self.gauge(metric_name, value, tags=tags, device_name=device_name)
+        else:
+            self.log.error('Metric type "%s" unknown' % (metric_type))
+
+    def _rest_request_to_json(self, address, object_path, service_name=JMX_SERVICE_CHECK, *args, **kwargs):
+        '''
+        Query the given URL and return the JSON response
+        '''
+        response_json = None
+
+        service_check_tags = ['instance:%s' % self.hostname]
+
+        url = address
+
+        if object_path:
+            url = self._join_url_dir(url, object_path)
+
+        # Add args to the url
+        if args:
+            for directory in args:
+                url = self._join_url_dir(url, directory)
+
+        # Add kwargs as arguments
+        if kwargs:
+            query = '&'.join(['{}={}'.format(key, value) for key, value in kwargs.iteritems()])
+            url = urljoin(url, '?' + query)
+
+        try:
+            response = requests.get(url)
+            response.raise_for_status()
+            response_json = response.json()
+
+        except Timeout as e:
+            self.service_check(service_name,
+                AgentCheck.CRITICAL,
+                tags=service_check_tags,
+                message="Request timeout: {0}, {1}".format(url, e))
+            raise
+
+        except (HTTPError,
+                InvalidURL,
+                ConnectionError) as e:
+            self.service_check(service_name,
+                AgentCheck.CRITICAL,
+                tags=service_check_tags,
+                message="Request failed: {0}, {1}".format(url, e))
+            raise
+
+        except JSONDecodeError as e:
+            self.service_check(service_name,
+                AgentCheck.CRITICAL,
+                tags=service_check_tags,
+                message='JSON Parse failed: {0}, {1}'.format(url, e))
+
```

Comment:
We should probably also re-raise the error in this case, or is there a reason not to do so?